.github/PULL_REQUEST_TEMPLATE.md

```diff
@@ -2,32 +2,41 @@
 
-1. Please write a description in this text box of the changes that are being
-   made.
+The following are the guidelines to help the review process go smoothly. Please read them carefully and fill out relevant information as much as possible.
 
-2. Please ensure that you have written units tests for the changes made/features
-   added.
-
-3. If you are closing an issue please use one of the automatic closing words as
-   noted here: https://help.github.com/articles/closing-issues-using-keywords/
+Many thanks in advance for your cooperation!
 
-4. If your pull request is not ready for review but you want to make use of the
-   continuous integration testing facilities please label it with `[WIP]`.
+-->
 
-5. If your pull request is ready to be reviewed without requiring additional
-   work on top of it, then remove the `[WIP]` label (if present).
+<!--
+Please replace #xxxx with the ID of the issue fixed in this PR. If such issue does not exist, please consider filing one and link it here.
+-->
+Fixes #xxxx.
 
-6. Once all work has been done and review has taken place please do not add
-   features or make changes out of the scope of those requested by the reviewer
-   (doing this just add delays as already reviewed code ends up having to be
-   re-reviewed/it is hard to tell what is new etc!). Further, please avoid
-   rebasing your branch during the review process, as this causes the context
-   of any comments made by reviewers to be lost. If conflicts occur during
-   review then they should be resolved by merging into the branch used for
-   making the pull request.
+### Description
 
-Many thanks in advance for your cooperation!
+<!--
+Please provide a description of the changes proposed in this pull request. Here are some questions to help you fill out the description:
 
+- What is the problem you are trying to solve? Describe it from the user's perspective. If you have an existing github issue, please add a summary of the issue here.
+- After this change, what will the user experience be like? Please describe any user-facing changes, such as new configurations or new behaviors.
+- How are you fixing the problem? Please provide a technical description of your solution. You can add or link your design doc if it exists.
+- How are the new features/behaviors tested? Please describe the test cases you added or modified. If they are tested in a cluster, please describe it as well.
 -->
+
+### Checklists
+
+<!-- Check the items below by putting "x" in the brackets for what is done. Not all of these items may be relevant to every PR. -->
+
+This PR has:
+
+- [ ] added documentation for new or modified features or behaviors.
+- [ ] updated the license in the source code files when it is required.
+- [ ] added new tests or modified existing tests to cover new code paths.
+      (Please explain in the PR description how the new code paths are tested, such as names of the tests that cover them.)
```

Comment:
```suggestion
      If there are new tests, please explain in the PR description how the new code paths are tested, such as names of the tests that cover them.
```

Reply:
I don't think this is applicable only when new tests are added. It is encouraged to explain the test coverage even if the new code paths are covered by existing tests.

Reply:
So, description about testing was asked [here](https://github.com/NVIDIA/spark-rapids/pull/13211/files#diff-18813c86948efc57e661623d7ba48ff94325c9b5421ec9177f724922dd553a35R26) first but is being asked again here, because it is important but people often forget it. I think this should suffice:

```
- [ ] added new tests or modified existing tests to cover new code paths.
      (Please explain in the PR description how the new code paths are tested, such as names of the new/existing tests that cover them.)
```